app/namespace-watcher.go

```diff
@@ -45,77 +41,149 @@ func createLimitRange(clientset *kubernetes.Clientset, namespaceName string) err
+	logrus.Info("Starting Namespace-Watcher with the following parameters:")
+	logrus.Info("CPU_LIMIT_MIN ", setLimits.CpuLimitMin)
+	logrus.Info("CPU_LIMIT_MAX ", setLimits.CpuLimitMax)
+	logrus.Info("MEM_LIMIT_MIN ", setLimits.MemLimitMin)
+	logrus.Info("MEM_LIMIT_MAX ", setLimits.MemLimitMax)
+	logrus.Info("EPHEMERAL_STORAGE_MIN ", setLimits.EphemeralStorageLimitMin)
+	logrus.Info("EPHEMERAL_STORAGE_MAX ", setLimits.EphemeralStorageLimitMax)
+
 	// create Kubernetes API client
 	config, err := rest.InClusterConfig()
 	if err != nil {
-		fmt.Println("Failed to get Kubernetes config:", err)
-		os.Exit(1)
+		logrus.Fatal("Failed to get Kubernetes config: ", err)
 	}
 	clientset, err := kubernetes.NewForConfig(config)
 	if err != nil {
-		fmt.Println("Failed to create Kubernetes client:", err)
-		os.Exit(1)
+		logrus.Fatal("Failed to create Kubernetes client: ", err)
 	}
 
 	// watch for namespace creation events
 	watcher, err := clientset.CoreV1().Namespaces().Watch(context.Background(), metav1.ListOptions{})
 	if err != nil {
-		fmt.Println("Failed to watch namespaces:", err)
-		os.Exit(1)
+		logrus.Fatal("Failed to watch namespaces: ", err)
+
+	}
+
+	// create a set to store excluded namespaces
+	excludedNamespaces := mapset.NewSet[string]()
+	excludedNamespaces.Add("default")
+	excludedNamespaces.Add("cattle")
+	excludedNamespaces.Add("kube-system")
+	excludedNamespaces.Add("kube-public")
+	excludedNamespaces.Add("istio-system")
+	excludedNamespaces.Add("kube-node-lease")
+	excludedNamespaces.Add("kube-local")
+
+	// read additional excluded namespaces from the environment variable
+	additionalExcluded := os.Getenv("EXCLUDED_NAMESPACES")
+	if additionalExcluded != "" {
+		additionalExcludedList := strings.Split(additionalExcluded, ",")
+		for _, ns := range additionalExcludedList {
+			excludedNamespaces.Add(strings.TrimSpace(ns))
+		}
 	}
 
 	// process events
 	for event := range watcher.ResultChan() {
 		// check if event is a namespace creation event
 		if event.Type == watch.Added {
-			// print namespace name to STDOUT
 			namespaceName := event.Object.(*corev1.Namespace).ObjectMeta.Name
 
-			//exclude cattle and system namespaces
-			if strings.Contains(namespaceName, "default") {
-				log.Printf("Skipping namespace %s\n", namespaceName)
-				continue
-			}
-			if strings.Contains(namespaceName, "cattle") || strings.Contains(namespaceName, "kube-system") || strings.Contains(namespaceName, "kube-public") {
-				log.Printf("Skipping namespace %s\n", namespaceName)
-				continue
-			}
-			if strings.Contains(namespaceName, "istio-system") || strings.Contains(namespaceName, "kube-public") || strings.Contains(namespaceName, "kube-local") {
-				log.Printf("Skipping namespace %s\n", namespaceName)
+			// check if namespace is in the excluded set or if it contains word "cattle"
+			isExcluded := strings.Contains(namespaceName, "cattle") || excludedNamespaces.Contains(namespaceName)
+
+			if isExcluded {
```

Comment:
The `cattle` namespace is already added to the list of excluded namespaces (see line 150).

This code can be simplified in this way:

```suggestion
			if excludedNamespaces.Contains(namespaceName) {
```

Reply:
I will address in rc1